src/app/hooks/finalityProviders/useFinalityProvidersData.ts

```diff
@@ -0,0 +1,91 @@
+import { useCallback, useEffect, useMemo, useState } from "react";
+
+import {
+  FinalityProvider as FinalityProviderInterface,
+  SortDirection,
+  SortField,
+} from "@/app/types/finalityProviders";
+
+export const useFinalityProvidersData = (
+  initialProviders: FinalityProviderInterface[] | undefined,
+) => {
+  const [filteredProviders, setFilteredProviders] = useState(initialProviders);
+  const [sortField, setSortField] = useState<SortField>("stakeSat");
+  const [sortDirection, setSortDirection] = useState<SortDirection>("desc");
+
+  useEffect(() => {
+    setFilteredProviders(initialProviders);
+  }, [initialProviders]);
+
+  const handleSearch = useCallback(
```

Comment:
Are we okay that if I sort by "Finality Provider" I will receive `no data` first, then FPs starting with the numbers? https://i.imgur.com/kSRWFOR.png
I think in terms of UX it's better to have `no data` at the end of the list when I search by name. @jrwbabylonlab @vitsalis what are your thoughts?